src/prefect/orion/services/cancellation_cleanup.py

```diff
@@ -0,0 +1,140 @@
+"""
+The CancellationCleanup service. Responsible for cancelling tasks and subflows that haven't finished.
+"""
+
+import asyncio
+
+import pendulum
+import sqlalchemy as sa
+from sqlalchemy.ext.asyncio import AsyncSession
+from sqlalchemy.sql.expression import or_
+
+import prefect.orion.models as models
+from prefect.orion.database.dependencies import inject_db
+from prefect.orion.database.interface import OrionDBInterface
+from prefect.orion.schemas import filters, states
+from prefect.orion.services.loop_service import LoopService
+from prefect.settings import PREFECT_ORION_SERVICES_CANCELLATION_CLEANUP_LOOP_SECONDS
+
+NON_TERMINAL_STATES = list(set(states.StateType) - states.TERMINAL_STATES)
+
+
+class CancellationCleanup(LoopService):
+    """
+    A simple loop service responsible for cancelling tasks and subflows left over from
+    cancelling flow runs
+    """
+
+    def __init__(self, loop_seconds: float = None, **kwargs):
+        super().__init__(
+            loop_seconds=loop_seconds
+            or PREFECT_ORION_SERVICES_CANCELLATION_CLEANUP_LOOP_SECONDS.value(),
+            **kwargs,
+        )
+
+        # query for this many runs to mark failed at once
+        self.batch_size = 200
+
+    @inject_db
+    async def run_once(self, db: OrionDBInterface):
+        """
+        - cancels active tasks belonging to recently cancelled flow runs
+        - cancels any active subflow that belongs to a cancelled flow
+        """
+        async with db.session_context(begin_transaction=True) as session:
+            # cancels active tasks belonging to recently cancelled flow runs
+            while True:
+                cancelled_flow_query = (
+                    sa.select(db.FlowRun)
+                    .where(
+                        db.FlowRun.state_type == states.StateType.CANCELLED,
+                        db.FlowRun.end_time != None,
+                        db.FlowRun.end_time >= (pendulum.now("UTC").subtract(days=1)),
+                    )
+                    .limit(self.batch_size)
+                )
+
+                flow_run_result = await session.execute(cancelled_flow_query)
+                flow_runs = flow_run_result.scalars().all()
+
+                for run in flow_runs:
+                    await self._cancel_child_runs(session=session, flow_run=run)
+
+                # if no relevant flows were found, exit the loop
```

Comment:
I _think_ this comment is wrong, seems like it should be saying something like "If we processed fewer flows than our batch size then no more flows are left to process, ext the loop.".